clients/filesystem-fuse/src/filesystem.rs

```diff
@@ -201,6 +246,86 @@ pub(crate) struct OpenedFile {
     pub writer: Option<Box<dyn FileWriter>>,
 }
 
+impl OpenedFile {
+    pub fn new(file_stat: FileStat) -> Self {
+        OpenedFile {
+            file_stat: file_stat,
+            handle_id: 0,
+            reader: None,
+            writer: None,
+        }
+    }
+
+    async fn read(&mut self, offset: u64, size: u32) -> Result<Bytes> {
+        let reader = self.reader.as_mut().ok_or(Errno::from(libc::EBADF))?;
+        let result = reader.read(offset, size).await?;
+
+        // update the atime
+        self.file_stat.atime = Timestamp::from(SystemTime::now());
+
+        Ok(result)
+    }
+
+    async fn write(&mut self, offset: u64, data: &[u8]) -> Result<u32> {
+        let writer = self.writer.as_mut().ok_or(Errno::from(libc::EBADF))?;
+        let written = writer.write(offset, data).await?;
```

Comment:
please handle the cases `written` is less than data.len()

Reply:
Why need to handle that case here?

Reply:
seems a bug here,
```rust
        let end = offset + data.len() as u64;
```

Reply:
fix